src/crimson/osd/osd.cc

```diff
@@ -352,6 +337,9 @@ seastar::future<> OSD::ms_dispatch(ceph::net::ConnectionRef conn, MessageRef m)
   switch (m->get_type()) {
   case CEPH_MSG_OSD_MAP:
     return handle_osd_map(conn, boost::static_pointer_cast<MOSDMap>(m));
```

Comment:
after change app.run_deprecated to app.run, the app will automatically exit when get CEPH_MSG_MON_GET_VERSION_REPLY message,  so need explicitly execute shutdown() for those messengers otherwise   there is some problem.